src/context/product-context.js

```diff
@@ -0,0 +1,30 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import axios from "axios";
+import { initialProduct, productReducer } from "../Reducers/productReducer";
+import { ACTIONS } from "../Constants/actions";
+const productContext = createContext();
+
+const ProductProvider = (props) => {
+  const [productState, productDispatch] = useReducer(
+    productReducer,
+    initialProduct
+  );
+  useEffect(
+    () =>
+      (async () => {
+        const { data } = await axios.get("/api/products");
+        productDispatch({
+          type: ACTIONS.DISPLAY_PRODUCTS,
+          payload: { data: data.products },
+        });
+      })(),
+    []
+  );
```

Comment:
i don't think you should make axios call in context because whenever you load your app every time this call will be made not when it is needed